app/components/action_button.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# Draws a blue button in the side menu
+class ActionButton < ApplicationComponent
+  # @param [Hash] properties the button properties
+  def initialize(label:, url:, confirm: nil, new_page: nil, check_url: nil, disabled: nil)
+    @label = label
```

Comment:
it feels like there ought to be a more elegant way to avoid so much boilerplate code like this

Reply:
We could make a struct, but that's just moving the boilerplate, isn't it?